### ISSUE TITLE
[DOC] Add type hints for shapelet-based collection transformation.

### ISSUE THREAD

**Issue by baraline** [aeon-toolkit/aeon#1910]
### Describe the issue

Other modules are increasingly using type hints in the function declaration, the shapelet-based tranformation module should do the same.

### Suggest a potential alternative/fix

Add type hints to algorithms in `aeon/transformations/collection/shapelet_based`. One PR can be opened for each algorithm if preferred, as some are quite large. 

### Additional context

If you have any question, don't hesitate to ask directly here or contact us on Slack!

**Comment by baraline** [comment]
@Datadote If this one is OK for you, you can assign yourself to it !

**Comment by Datadote** [comment]
Aeon-Assign bot assign @Datadote

**Comment by baraline** [comment]
I don't remember if we have a bot for this, but you can do it manually on the Assignees tab on the right side, I'll do it for you this time. Happy coding, and thanks for contributing !

**Comment by MatthewMiddlehurst** [comment]
There was an error in the docs that we have now fixed, but have to wait for a release for it to appear on the stable branch. Have to start with @aeon-actions-bot

**Comment by MatthewMiddlehurst** [comment]
I think you need write access to manually assign yourself

**Comment by Datadote** [comment]
Yeah, I don't access to manually assign myself (Tried on other issues).

I started a draft pull request with type hints, and some styling questions in #1917 . Thank you for the help

**Comment by aryanpola** [comment]
Aeon-Assign bot assign @aryanpola
